src/app/public/modules/datepicker/datepicker-input-fuzzy.directive.ts

```diff
@@ -251,13 +265,21 @@ export class SkyFuzzyDatepickerInputDirective
 
   constructor(
     private changeDetector: ChangeDetectorRef,
-    private elementRef: ElementRef,
-    private renderer: Renderer2,
-    @Optional() private datepickerComponent: SkyDatepickerComponent,
     private configService: SkyDatepickerConfigService,
+    private elementRef: ElementRef,
     private fuzzyDateService: SkyFuzzyDateService,
-    private resourcesService: SkyLibResourcesService
-  ) { }
+    private localeProvider: SkyAppLocaleProvider,
+    private renderer: Renderer2,
+    private resourcesService: SkyLibResourcesService,
+    @Optional() private datepickerComponent: SkyDatepickerComponent
+  ) {
+    this.localeProvider.getLocaleInfo()
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe((localeInfo) => {
+        moment.locale(localeInfo.locale || 'en-US');
+        this.localeProviderDateFormat = moment.localeData().longDateFormat('L');
```

Comment:
It looks like this same locale logic is being used in three different places. Is there a universal location we could put this code so that if we fix it in one place, all three directives benefit?